lib/posix/Kconfig.sched

```
@@ -0,0 +1,10 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config POSIX_SCHED
+	bool "_POSIX_PRIORITY_SCHEDULING API support"
+	default y if PTHREAD
+	depends on PTHREAD
+	help
+	  This enables POSIX scheduling APIs (_POSIX_PRIORITY_SCHEDULING).
```

I really ❤️  where this is going. Basically where I was headed too. Two minor adjustments, if that's ok, because I'm hoping to try and make POSIX Kconfig option names map directly to POSIX Options / Groups.

In the future, it might change _slightly_. Can chat a bit about it on Discord if you like.
```suggestion
config POSIX_PRIORITY_SCHEDULING
	bool "_POSIX_PRIORITY_SCHEDULING API support"
	default y if PTHREAD
	default y if POSIX_API
	depends on PTHREAD
	help
	  This enables POSIX scheduling APIs (_POSIX_PRIORITY_SCHEDULING).
```

---

ok, my initial intention was also to start using POSIX names here